architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,300 @@
+
+The Kubernetes API Server, with which Applications RP interacts, runs in the `kube-system` namespace within the cluster.
+
+Today, the RP assumes that incoming requests are from a trusted user who is authorized to make the request. It should enforce this by checking that the requests are made by UCP. UCP in turn should accept requests from authorized and authenticated users and clients such as `rad cli`. Once we have Radius RBAC support, we should configure UCP so that it accepts requests only from authorized and authenticated users.
+
+Applications RP deploys each Application and its resources in its own namespace. This is necessary for administrators of the Kubernetes cluster to correctly configure RBAC and limit the scope of permissions. 
+
+## Assumptions 
+
+This threat model assumes that:
+
+1. The Radius installation is not tampered with.
+2. The Kubernetes cluster that Radius is installed on is not compromised.
+3. It is the responsibility of the Kubernetes cluster to authenticate users. Administrators and users with sufficient privileges can perform their required tasks. Radius cannot prevent actions taken by an administrator.
+4. Radius stores and queues used by Applications RP are not compromised. 
+HTTP requests forwarded to Applications RP from UCP originated from an authorized user. HTTP requests or other network traffic that has not been forwarded by UCP are considered unauthorized.
+
+## Data Flow
+
+![Applications RP via Microsoft Threat Modeling Tool](2024-10-applications-rp-threat-model/apprp-dataflow.png)
+
+
+Below are the key points associated with data flow:
+1. Applications RP receives request to deploy resources from UCP and sends back appropriate response.
+2. Depending on the resource to be deployed Applications RP takes the next step 
+    1. If the request is to deploy a recipe, Applications RP first downloads the recipe from a OCI registry (for bicep) or public module (for terraform)
+    2. Applications RP requests UCP to deploy bicep recipes. Or if its a Terraform recipe, the downloaded recipe is executed using the installed terraform, in the empty terraform directory mounted into the Applications RP pod. As part of this execution, Terraform integrated in RP makes AWS and Azure API calls. 
+    3. If it is a kubernetes resource such as container, the RP interacts with API server to render the resource.
+3. Application RP uses API server to save Radius resources and Async Operations By default, API Server is used as Radius Datastore and Queue.
+
+
+## Threats
+
+#### Impersonating Applications RP could cause information disclosure, DDoS and misuse of cloud resources.
+
+**Description:** 
+If a malicious actor can impersonate Applications RP, requests from UCP will be received by the malicious actor. The malicious actor can also send requests to UCP and access cloud credentials.
```

Comment:
> The malicious actor can also send requests to UCP and access cloud credentials.

If this is true, it's a different threat right? This is about intercepting traffic sent from UCP -> Applications RP.

Reply:
broke this into two threats.